### PR TITLE
Updates to build static clusterctl 

### PR DESCRIPTION
**Pull request by akutz**
**What this PR does / why we need it**:
This patch updates clusterctl targets to build statically linked binaries.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
I was building `clusterctl` for releases manually, and this way we can rely on the Docker build for those too.

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by akutz** [comment]
@andrewsykim I updated this to build the normal clusterctl as statically linked as well.

File: Makefile [hunk 1]
```diff
@@ -42,13 +42,18 @@ manager: fmt vet
...
 
 clusterctl-in-docker:
 	docker run --rm -v $(CWD):/go/src/sigs.k8s.io/cluster-api-provider-vsphere \
 	  -w /go/src/sigs.k8s.io/cluster-api-provider-vsphere \
-	  -e GOOS -e GOHOSTOS golang:1.12 \
-	  go build -o bin/clusterctl ./cmd/clusterctl
+	  -e CGO_ENABLED=0 -e GOOS="$${GOOS:-linux}" -e GOARCH="$${GOARCH:-amd64}" \
+	  golang:1.12 \
+	  go build -ldflags '-extldflags "-static" -w -s' \
+	  -o bin/clusterctl."$${GOOS:-linux}"_"$${GOARCH:-amd64}" ./cmd/clusterctl
```
Comment (andrewsykim) [inline Makefile]: The bin name changing is going to break some docs and blogs

Comment (akutz) [reply Makefile]: Then I'll add the same hack I have in the normal target. We should always include the arch in a Docker output. This is really just fixing that. But I'll add that hack for now.

Comment (akutz) [reply Makefile]: Actually, the clusterctl binaries are on the releases pages now. It should be grab them from there. If a person wants to build it locally then they should be fine handling this IMO.

Comment (akutz) [reply Makefile]: Fixed for now. We should change this later. This target shouldn't be in the Makefile IMO.

File: Makefile [hunk 2]
```diff
@@ -42,13 +42,19 @@ manager: fmt vet
 
 # Build the clusterctl binary
 clusterctl: fmt vet
-	go build -o bin/clusterctl ./cmd/clusterctl
+	CGO_ENABLED=0 go build -ldflags '-extldflags "-static" -w -s' \
+	-o bin/clusterctl."$${GOOS:-$$(go env GOHOSTOS)}"_"$${GOARCH:-$$(go env GOHOSTARCH)}" \
```
Comment (andrewsykim) [inline Makefile]: The GOOS and GOHOSTARCH overrides doesn't seem necessary for this case?

Comment (andrewsykim) [reply Makefile]: oh nvm this is just the bin name

Comment (akutz) [reply Makefile]: /hold

Comment (akutz) [reply Makefile]: Wait, never mind. Thought I saw a bug. There's not.

Comment (akutz) [reply Makefile]: /hold cancel

**Pull request merged**